File: mobile/native_mobile/api/src/main/java/com/domestikapp/api/mapper/AuthMapper.kt
package com.domestikapp.api.mapper

import com.domestikapp.api.models.AuthResponse
import com.domestikapp.api.models.NewUserPayload
import com.domestikapp.domain.model.Auth
import com.domestikapp.domain.model.User


fun AuthResponse.toDomainModel(): Auth {
    return Auth( jwtToken = this.jwtToken, refreshToken = this.refreshToken)
}

File: mobile/flutter_mobile/api/lib/mapper/AuthMapper.dart
import 'package:auto_mappr_annotation/auto_mappr_annotation.dart';
import 'package:domestikapp_api/api-responses/AuthResponse.dart';
import 'package:domestikapp_api/mapper/AuthMapper.auto_mappr.dart';
import 'package:domestikapp_domain/models/auth.dart';


@AutoMappr([
  MapType<AuthResponse, AuthModel>(
    fields: [
      Field('jwtToken', from: 'jwtToken'),
      Field('refreshToken', from: 'refreshToken'),
    ],
  ),
])
class AuthMapper extends $AuthMapper{}
